compiler-rt/lib/scudo/standalone/combined.h

```diff
@@ -1546,11 +1547,17 @@ class Allocator {
         RBEntryStart)[N];
   }
 
-  void mapAndInitializeRingBuffer() {
-    if (getFlags()->allocation_ring_buffer_size <= 0)
+  void initRingBufferMaybe() {
+    static HybridMutex RingBufferLock;
+    ScopedLock L(RingBufferLock);
```

Comment:
I'm thinking the case mentioned [here](https://github.com/llvm/llvm-project/pull/79670#discussion_r1469997581), does that mean we can't put this lock as static local variable?

Reply:
Yeah, I made it global and and added it to the enable/disable code.

There is a chance that if someone forks while the code holds the init lock, but before the ring buffer is finished, trying to init in the forked process would deadlock.